lms/djangoapps/teams/views.py

```diff
@@ -368,6 +368,9 @@ def post(self, request):
             }, status=status.HTTP_400_BAD_REQUEST)
         else:
             team = serializer.save()
+            if not has_access(request.user, 'staff', course_key):
```

Comment:
I'll double check with Leslie tomorrow, but I was under the impression that we didn't want to cover the third option, which is why this PR doesn't cover it. If we want to support that case, I'd be happy to add it in.